Cython/Compiler/ExprNodes.py

```diff
@@ -6632,16 +6632,68 @@ class PyMethodCallNode(CallNode):
     # arg_tuple   TupleNode     the arguments for the args tuple
     # kwdict      ExprNode or None  keyword dictionary (if present)
     # unpack      bool
+    # function_obj ExprNode or None   copy of self.function.obj (in some situations)    
 
-    subexprs = ['function', 'arg_tuple', 'kwdict']
+    subexprs = ['function', 'arg_tuple', 'kwdict', 'function_obj']
     is_temp = True
     kwdict = None
+    direct_fallback = False
+    function_obj = None
+
+    def __init__(self, pos, **kwds):
+        super().__init__(pos, **kwds)
+        if self.allow_direct_fallback_call():
+            self.direct_fallback = True
+            self.function_obj = ProxyNode(self.function.obj)
+            self.function.obj = CloneNode(self.function_obj)
+
+    def allow_direct_fallback_call(self):
+        if self.kwdict and not isinstance(self.kwdict, DictNode):
+            return False
+        if not self.function.is_attribute:
+            return False
+        return True
+    
+    def generate_direct_fallback_guard(self, code, negate):
+        if self.direct_fallback:
+            # negate gives the guard for the regular code
+            code.putln(f"#if {'!' if negate else ''}(CYTHON_VECTORCALL && !CYTHON_UNPACK_METHODS)")
+
+    def generate_direct_fallback(self,  code, len_args, keyword_variable):
```

Comment:
The fact that the calling code has a ready-made "keyword_variable" with a leading comma separator seems a pure accident. I'd rather pass the args and kwdict and then do a proper code generation here. Calling code shouldn't know too much about the way the code is generated here.